### PR TITLE
core: record store approximate size 

### PR DESCRIPTION
**Pull request by Connor1996**
use Mb as unit, add RegionSize and LeaderSize in StoreInfo

### CODE REVIEW

File: server/cache.go [hunk 1]
```diff
@@ -384,6 +386,9 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
 		if len(origin.DownPeers) > 0 || len(origin.PendingPeers) > 0 {
 			saveCache = true
 		}
+		if region.ApproximateSize != origin.ApproximateSize {
+			saveCache = true
```
Comment (siddontang) [inline server/cache.go]: do we need save here? 

@disksing

File: server/cache_test.go [hunk 2]
```diff
@@ -444,8 +444,8 @@ func (s *testClusterInfoSuite) testRegionHeartbeat(c *C, cache *clusterInfo) {
 	}
 
 	for _, store := range cache.Stores.GetStores() {
-		c.Assert(store.LeaderCount, Equals, cache.Regions.GetStoreLeaderCount(store.GetId()))
-		c.Assert(store.RegionCount, Equals, cache.Regions.GetStoreRegionCount(store.GetId()))
+		c.Assert(store.LeaderSize, Equals, cache.Regions.GetStoreLeaderRegionSize(store.GetId()))
+		c.Assert(store.RegionSize, Equals, cache.Regions.GetStoreRegionSize(store.GetId()))
```
Comment (disksing) [inline server/cache_test.go]: I suppose it's better to check both count and size.

File: server/metrics.go [hunk 3]
```diff
@@ -96,6 +96,14 @@ var (
...
 			Help:      "Counter of tso events",
 		}, []string{"type"})
+
+	storeStatusGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "pd",
+			Subsystem: "schedule",
+			Name:      "store_status",
+			Help:      "Store status for schedule",
+		}, []string{"store", "type"})
```
Comment (disksing) [inline server/metrics.go]: Make it `namespace, store, type`, we need filter stores by namespace.

File: server/store_statistics.go [hunk 4]
```diff
@@ -73,6 +74,12 @@ func (s *storeStatistics) Observe(store *core.StoreInfo) {
 	id := strconv.FormatUint(store.GetId(), 10)
 	balanceScoreGauge.WithLabelValues(s.namespace, id, "region").Set(store.RegionScore())
 	balanceScoreGauge.WithLabelValues(s.namespace, id, "leader").Set(store.LeaderScore())
+
+	storeID := fmt.Sprintf("store_%d", store.GetId())
+	storeStatusGauge.WithLabelValues(storeID, "region_size").Set(float64(store.RegionSize))
+	storeStatusGauge.WithLabelValues(storeID, "region_count").Set(float64(store.RegionCount))
+	storeStatusGauge.WithLabelValues(storeID, "leader_size").Set(float64(store.LeaderSize))
+	storeStatusGauge.WithLabelValues(storeID, "leader_count").Set(float64(store.LeaderCount))
```
Comment (disksing) [inline server/store_statistics.go]: Maybe better to merge `balanceScoreGauge` with it.
BTW, we usually use only id as metrics label and config grafana to show the label as `tikv-{{id}}`.

File: server/cache.go (hunk 1 above)
Comment (disksing) [reply server/cache.go]: Yes. Schedulers need it update in time.

Comment (siddontang) [reply server/cache.go]: so I guess every heartbeat will save to etcd, this may reduce the performance.

Comment (disksing) [reply server/cache.go]: They are only saved to memory cache, and they won't update every time. We store region size in MB, please see https://github.com/pingcap/pd/pull/814/files#diff-1f3079039053bbe65b3d8b75ab87d5ceR325

**Comment by disksing** [comment]
LGTM. Don't forget to update grafana metrics.

File: server/core/region.go [hunk 5]
```diff
@@ -366,6 +380,22 @@ func (r *RegionsInfo) GetRegions() []*RegionInfo {
...
+}
+
+// GetStoreFollowerRegionSize get total size of store's follower regions
+func (r *RegionsInfo) GetStoreFollowerRegionSize(storeID uint64) uint64 {
+	return r.followers[storeID].TotalSize()
+}
+
+// GetStoreRegionSize get total size of store's regions
+func (r *RegionsInfo) GetStoreRegionSize(storeID uint64) uint64 {
+
```
Comment (siddontang) [inline server/core/region.go]: remove the empty line

**Pull request merged**